### DISCUSSION TITLE
Should SwitchStatement.cases contain a seperate case for common cases

### DISCUSSION THREAD

**Discussion by Yash-Singh1** [babel/babel#13670] [Q&A] [answered]
Try running in https://astexplorer.net/ with the parser set to `babel-eslint`:

```js
/* Brought from w3schools */

switch (new Date().getDay()) {
  case 4:
  case 5:
    text = "Soon it is Weekend";
    break;
  case 0:
  case 6:
    text = "It is Weekend";
    break;
  default:
    text = "Looking forward to the Weekend";
}
```

Note how the cases property underneath the `SwitchStatement` node contains 5 cases while there are only 3, two of them being common with another. Is this expected?

**Comment by fedeci** [comment, marked as answer]
The fall through cases can be distinguished by the empty `consequent` property.
They are still switch cases even if they are in common with other ones so a separate node wouldn't make sense. :)
